lk2nd/partial-goods/partial-goods.c

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/* Copyright (c) 2022, Otto Pflüger <affenull2345@gmail.com> */
+
+#include <platform/partial_goods.h>
+#include <boot.h>
+#include <lk2nd/util/lkfdt.h>
+
+#if TARGET_MSM8909
+static const char *cpu_disable_paths[] = {
+	"/cpus/cpu@2",
+	"/soc@0/power-manager@b0a8000",
+	"/soc@0/power-manager@b0a9000",
+	"/cpus/cpu@3",
+	"/soc@0/power-manager@b0b8000",
+	"/soc@0/power-manager@b0b9000",
```

Comment:
Also, if we special case the disabling of CPUs it should be quite easy to check the `enum boot_type` for BOOT_DOWNSTREAM and replace the `device_type` like in the original implementation instead. That way it would also work for downstream if there is some msm-4.9 device with MSM8905 out there. This is optional though.

Would you like to try implementing this? I can also try to make these changes if you prefer to focus on other things.